cmd/snapshotgc/snapshotgc.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	applicationapiv1alpha1 "github.com/redhat-appstudio/application-api/api/v1alpha1"
+	releasev1alpha1 "github.com/redhat-appstudio/release-service/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	scheme = runtime.NewScheme()
+)
+
+func init() {
+	applicationapiv1alpha1.AddToScheme(scheme)
+	clientgoscheme.AddToScheme(scheme)
+	releasev1alpha1.AddToScheme(scheme)
+}
+
+type snapshotData struct {
+	snapshot applicationapiv1alpha1.Snapshot
+	// environmentBinding applicationapiv1alpha1.SnapshotEnvironmentBinding
+	release releasev1alpha1.Release
+}
+
+func getSnapshotsForReleases(
+	cl client.Client,
+	snapToData map[string]snapshotData,
+) map[string]snapshotData {
```

Comment:
`map[string]snapshotData`

I'd like to know little bit more about this structure, why it's uses `namespace|name` as a key, will it be used by future function for lookups?

Reply:
> why it's uses namespace|name as a key
I was hoping to use the snapshot itself as key, but I was not able to make it work (not comparable or something like that), so I opted for a string, and this combination is something I expect to be unique.
I'm open for other suggestions.

> will it be used by future function for lookups?
Yes.

Reply:
The logic will be per-namespace, so I changed it to be only the snapshot name.